Content.Client/Doors/Electronics/DoorElectronicsBoundUserInterface.cs

```diff
@@ -0,0 +1,62 @@
+using Content.Shared.Access;
+using Content.Shared.Doors.Electronics;
+using Robust.Client.GameObjects;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.Doors.Electronics;
+
+public sealed class DoorElectronicsBoundUserInterface : BoundUserInterface
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+
+    private DoorElectronicsConfigurationMenu? _window;
+
+    public DoorElectronicsBoundUserInterface(EntityUid owner, Enum uiKey) : base(owner, uiKey)
+    {
+    }
+
+    protected override void Open()
+    {
+        base.Open();
+        List<string> accessLevels = new();
+
+        foreach (var accessLevel in _prototypeManager.EnumeratePrototypes<AccessLevelPrototype>())
+        {
+            if (accessLevel.Name != null)
+            {
+                accessLevels.Add(accessLevel.ID);
+            }
+        }
+
+        accessLevels.Sort();
+
+        _window = new DoorElectronicsConfigurationMenu(this, accessLevels, _prototypeManager);
+        _window.OnClose += Close;
+        _window.OpenCentered();
+
+        SendMessage(new DoorElectronicsRefreshUiMessage());
```

Comment:
Don't send a message, just have the server update the UI state on opening if its required.